src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2742,6 +2742,36 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
             $container->removeDefinition('mailer.messenger_transport_listener');
         }
 
+        if ($config['dkim_signer']['enabled']) {
+            $dkimSigner = $container->getDefinition('mailer.dkim_signer');
+            $dkimSigner->setArgument(0, $config['dkim_signer']['key']);
+            $dkimSigner->setArgument(1, $config['dkim_signer']['domain']);
+            $dkimSigner->setArgument(2, $config['dkim_signer']['select']);
+            $dkimSigner->setArgument(3, $config['dkim_signer']['options']);
+            $dkimSigner->setArgument(4, $config['dkim_signer']['passphrase']);
+
+            $dkimSignerListener = $container->getDefinition('mailer.dkim_signer.listener');
+            $dkimSignerListener->setArgument(0, $dkimSigner);
```

Comment:
this is useless, as it is already set in the config file.

Reply:
@stof  I have not been able to get it to work without this line (or the new replaceArgument I'm now using). It always complains with this message:`The service "mailer.dkim_signer.listener" has a dependency on a non-existent service "mailer.dkim_signer"`

Reply:
@stof I have removed these lines as you mentioned